### PR TITLE
btop: init at 1.0.5

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Snazzy resource viewer: https://github.com/aristocratos/btop/blob/main/Img/normal.png

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139018` at 703ab9f1 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>btop</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/system/btop/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by figsoda** [comment]
looks like this line is causing aarch64-linux build to fail
https://github.com/aristocratos/btop/blob/v1.0.5/Makefile#L82

**Comment by rmcgibbo** [comment]
Interesting. I didn't realize the default version of gcc is different on aarch64-linux.

File: pkgs/tools/system/btop/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-pa9i65ndx8LMTMRXyL2GXCauM6Q8gAb16zGOylQFwL0=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    make install PREFIX=$out
+    runHook postInstall
+  '';
```
Comment (figsoda) [inline pkgs/tools/system/btop/default.nix:21]:
```suggestion
  installFlags = [ "PREFIX=$out" ];
```

File: pkgs/tools/system/btop/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "A monitor of resources";
+    homepage = "https://htop.dev";
```
Comment (figsoda) [inline pkgs/tools/system/btop/default.nix:25]:
```suggestion
    homepage = "https://github.com/aristocratos/btop";
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/btop/default.nix (hunk 1 above)
Comment (rmcgibbo) [reply pkgs/tools/system/btop/default.nix:21]: For some reason this doesn't work. The result/ directory is empty when I build it this way.

Comment (figsoda) [reply pkgs/tools/system/btop/default.nix:21]: my bad, i forgot the parenthesis, edited the diff

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 139018` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>btop</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**